index.js

```diff
@@ -70,6 +70,17 @@ Router.prototype.on = function on (method, path, opts, handler, store) {
   // handler validation
   assert(typeof handler === 'function', 'Handler should be a function')
 
+  // path ends with optional parameter
+  const optionalParamRegex = /(\/:[^/]*?)\?(\/?)$/
```

Comment:
Out of curiosity, why the need for a regex instead of `path.endsWith('?')`

Reply:
I had `path.endsWith('?') || path.endsWith('?/')` before, but then it's not ensured to be a parameter.
That would also match for routes like `/foo?`, don't know if that is an issue..

Reply:
Gotcha! 👍